src/language/detect_language.py

```diff
@@ -70,7 +71,7 @@ def _build_ext_lang_map():
     return _ext_lang
 
 
-def detect_language(file_path):
+def detect_language(file_path, repo_dir=""):
```

Comment:
I think it would be better if there is only one parameter `file_path` which contains the `repo_dir` too. So the `detect_language` always gets the full file path.